UndertaleModLib/UndertaleData.cs

```diff
@@ -482,6 +437,46 @@ private bool TestGMS1Version(uint stableBuild, uint betaBuild, bool allowGMS2 =
             return (allowGMS2 || !IsGameMaker2()) && (IsVersionAtLeast(1, 0, 0, stableBuild) || (IsVersionAtLeast(1, 0, 0, betaBuild) && !IsVersionAtLeast(1, 0, 0, 1000)));
         }
 
+        /// <summary>
+        /// Sets the GMS2+ version flag in GeneralInfo.
+        /// </summary>
+        /// <param name="major">The major version.</param>
+        /// <param name="minor">The minor version.</param>
+        /// <param name="release">The release version.</param>
+        /// <param name="build">The build version.</param>
+        public void SetGMS2Version(uint major, uint minor = 0, uint release = 0, uint build = 0)
```

Comment:
`UndertaleGeneralInfo.GMSVersions.GMS2_2_2_302` qualifies as too much boilerplate, IMO.

Most invalid versions are actually just versions that don't have any distinguishing characteristics in the data file, but I see what you mean about major versions in particular. I can add a `throw` there.

Reply:
> UndertaleGeneralInfo.GMSVersions.GMS2_2_2_302 qualifies as too much boilerplate, IMO.

AFAIK, you don't have give the full name like that, `GMSVersions.GMS2_2_2_302` should be valid too, but may need some specific usings.

Reply:
Added error: "Attempted to set a version of GameMaker {major} using SetGMS2Version". Not going to resolve this in case it's still possible to get in those usings and make it pretty.